api/src/main/java/io/opencensus/stats/NoopStats.java

```diff
@@ -140,6 +150,12 @@ public void record() {}
     @Override
     public void record(TagContext tags) {
       Utils.checkNotNull(tags, "tags");
+
+      if (hasUnsupportedValues) {
+        // drop all the recorded values
+        logger.log(Level.WARNING, "Dropping values, value to record must be non-negative.");
+        hasUnsupportedValues = false;
```

Comment:
Why do you reset this value? If they call record twice they get values recorded?

Reply:
I thought will allow users to record values again, anyways removed the value reset.